### PR TITLE
Use correct clang_rt path in merge_build_host: false

### PR DESCRIPTION
**Pull request by xhochy**
This was only yet tested on Windows using R packages and thus this didn't fail. Tested now with `arrow-cpp` and found at least this as a problem.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
Isn't `BUILD_PREFIX` defined even with `merge_build_host: True`?

**Comment by xhochy** [comment]
> Isn't `BUILD_PREFIX` defined even with `merge_build_host: True`?

It is probably but we still need to check what the correct path to use is to also make this activation script pass in the test environment.

File: recipe/activate-clang_win-64.bat [hunk 1]
```diff
@@ -10,6 +20,6 @@ set "NM=llvm-nm.exe"
 set "LD=lld-link.exe"
 
 set "CPPFLAGS_USED=-D_CRT_SECURE_NO_WARNINGS -D_MT -D_DLL -nostdlib -Xclang --dependent-lib=msvcrt -fuse-ld=lld -fno-aligned-allocation"
-set "LDFLAGS=-nostdlib -Xclang --dependent-lib=msvcrt -fuse-ld=lld -Wl,-defaultlib:%PREFIX%/lib/clang/@PKG_VERSION@/lib/windows/clang_rt.builtins-x86_64.lib"
+set "LDFLAGS=-nostdlib -Xclang --dependent-lib=msvcrt -fuse-ld=lld -Wl,-defaultlib:%CLANG_BUILD_PREFIX%/lib/clang/@PKG_VERSION@/lib/windows/clang_rt.builtins-x86_64.lib"
```
Comment (isuruf) [inline recipe/activate-clang_win-64.bat:23]:
```suggestion
set "LDFLAGS=-nostdlib -Xclang --dependent-lib=msvcrt -fuse-ld=lld -Wl,-defaultlib:%CONDA_PREFIX%/lib/clang/@PKG_VERSION@/lib/windows/clang_rt.builtins-x86_64.lib"
```

Comment (xhochy) [reply recipe/activate-clang_win-64.bat:23]: Is there any documentation about all these prefixes? I would like to understand them more before moving to this again.

Comment (xhochy) [reply recipe/activate-clang_win-64.bat:23]: Ah, figured. This is defined by the order of environment activations; will test and then push.

**Comment by xhochy** [comment]
@isuruf tested and updated.

**Pull request merged**